### PR TITLE
bn: fix BN.divmod

### PR DESCRIPTION
**Pull request by fanatid**
See #96 for details

### CODE REVIEW

**Comment by fanatid** [comment]
`BN.ineg` also was fixed.

**Comment by indutny** [comment]
Needs test case. ;)

File: lib/bn.js [hunk 1]
```diff
@@ -2281,51 +2280,45 @@
...
       if (mode !== 'mod') {
-        div = res.div.neg();
+        res.div.ineg();
       }
 
       if (mode !== 'div') {
-        mod = res.mod.neg();
-        if (positive && mod.neg) {
-          mod = mod.add(num);
+        res.mod.ineg();
```
Comment (indutny) [inline lib/bn.js]: I don't think that this is correct, `res.mod` could be `=== this`.

Comment (fanatid) [inline lib/bn.js]: You right, I forgot about this! Updated.

**Comment by indutny** [comment]
Landed and released, thank you!

**Comment by indutny** [comment]
Landed and released, thank you!

**Pull request closed without merging**